lib/spack/spack/config.py

```diff
@@ -861,13 +935,19 @@ def _process_config_path(path):
 #
 # Settings for commands that modify configuration
 #
-def default_modify_scope():
+def default_modify_scope(subscopes=True):
```

Comment:
concept of a subscope needs to be documented somewhere -- I think we've only really called them platform scopes so far.

Reply:
I've switched to calling them platform scopes.